libtmux/common.py

```diff
@@ -374,10 +375,16 @@ def get_by_id(self, id):
 
 
 def which(
-    exe=None,
-    default_paths=["/bin", "/sbin", "/usr/bin", "/usr/sbin", "/usr/local/bin"],
-    append_env_path=True,
-):
+    exe: str,
+    default_paths: list[str] = [
```

Comment:
Needs 3.9: https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections

we can either wait until Monday (when i drop python 3.7 and 3.8 and you can rebase) or use `typing.List`

Reply:
if you change is to `typing.List` and CI passes I can merge this immediately

Reply:
Sure, no problem! Is there a style preference between the options below?

```python
# Import all type hints at top that will be used in this file, may be very long
from typing import Optional, List, Union, Any

def sample(array: List[Any], id: Union[str, int]) -> Optional[int]:
    ...
    return None
```

```python
# Use an alias to avoid explicitly importing types
import typing as T

def sample(array: T.List[T.Any], id: T.Union[str, int]) -> T.Optional[int]:
    ...
    return None
```

Reply:
I realize that I took the first approach by importing `Union` and `Optional` but that should be a quick fix if the second style is preferred.

Reply:
```
# Use an alias to avoid explicitly importing types
import typing as T
```

Make the `T` lowercase and that would be good

https://cs.github.com/?scopeName=All+repos&scope=&q=%22import+typing+as+T%22

Seems flask does it. https://github.com/pallets/flask/blob/cb7cd1e79b280d3fd80978ca82ae52881525a476/src/flask/blueprints.py